automata/regex/parser.py

```diff
@@ -216,6 +216,38 @@ def option(self):
         self._initial_state = new_initial_state
         self._final_states.add(new_initial_state)
 
+    def shuffle(self, other):
+        """
+        Apply the shuffle operation to the NFA represented by this builder and other.
+        No need for BFS since all states are acessible.
+        """
+        new_state_name_dict = dict()
+
+        def get_state_name(state_name):
+            return new_state_name_dict.setdefault(state_name, self.__get_next_state_name())
+
+        self._initial_state = get_state_name((self._initial_state, other._initial_state))
+
+        new_transitions = dict()
```

Comment:
@eliotwrobson I see a lot of references in the codebase to `dict()` instead of `{}`. Is there a reason for this, besides to possibly distinguish it from an empty `set()`?

Reply:
Yes, it's mainly just a habit of mine since I think it's less ambiguous than using `{}`. Fine switching styles though, if you'd like to keep that uniform through this project.

Reply:
@eliotwrobson "Up to 6 times longer"?? Eek that's good to know. I guess the StackOverflow has spoken once again!

!["I Have Spoken" from The Mandalorian](https://user-images.githubusercontent.com/872474/206037026-dd388a26-695b-4799-b240-2d522fc132ae.gif)